cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -624,8 +625,10 @@ updateOutputTemplate
     delegCoin = case mDelegatedSupply of Nothing -> 0; Just amountDeleg -> fromIntegral totalSupply - unLovelace amountDeleg
     nonDelegCoin = fromIntegral totalSupply - delegCoin
 
-    distribute :: Integer -> Int -> [AddressInEra ShelleyEra] -> [(AddressInEra ShelleyEra, Lovelace)]
-    distribute funds nAddrs addrs = zip addrs (fmap Lovelace (coinPerAddr + remainder:repeat coinPerAddr))
+    distribute :: Integer -> Int -> [AddressInEra ShelleyEra] -> m [(AddressInEra ShelleyEra, Lovelace)]
+    distribute funds nAddrs addrs
+      | funds < 0 = throwError GenesisCmdNegativeInitialFunds
```

Comment:
We can return the exact amount to the user so they can either:
- Decrease the delegated supply
- Increase the total supply

by this amount.